src/Core/Repositories/EntityFramework/Queries/EventReadPageByProviderIdActingUserIdQuery.cs

```diff
@@ -0,0 +1,41 @@
+using System.Linq;
+using Bit.Core.Models.EntityFramework;
+using System;
+using Bit.Core.Models.Data;
+
+namespace Bit.Core.Repositories.EntityFramework.Queries
+{
+    public class EventReadPageByProviderIdActingUserIdQuery : IQuery<Event>
+    {
+        private readonly Guid _providerId;
+        private readonly Guid _actingUserId;
+        private readonly DateTime _startDate;
+        private readonly DateTime _endDate;
+        private readonly DateTime? _beforeDate;
+        private readonly PageOptions _pageOptions;
+
+        public EventReadPageByProviderIdActingUserIdQuery(Guid providerId, Guid actingUserId,
+                DateTime startDate, DateTime endDate, DateTime? beforeDate, PageOptions pageOptions)
+        {
+            _providerId = providerId;
+            _actingUserId = actingUserId;
+            _startDate = startDate;
+            _endDate = endDate;
+            _beforeDate = beforeDate;
+            _pageOptions = pageOptions;
+        }
+
+        public IQueryable<Event> Run(DatabaseContext dbContext)
+        {
+            var q = from e in dbContext.Events
+                where e.Date >= _startDate &&
+                (_beforeDate != null || e.Date <= _endDate) &&
+                (_beforeDate == null || e.Date < _beforeDate.Value) &&
+                e.ProviderId == _providerId &&
+                e.ActingUserId == _actingUserId
+                orderby e.Date descending
+                select e;
+            return q.Skip(0).Take(_pageOptions.PageSize);
```

Comment:
`.Skip(0)` as a hard-coded option? Not sure why page size is used other than a simple limit, in which case if actual paging won't be used through Skip, not sure why the `Skip()` method is ever called here?

Reply:
It's a copy of the `EventReadPageByOrganizationIdActingUserIdQuery`.